src/tests/TdcTime_Convert.cpp

```diff
@@ -0,0 +1,143 @@
+namespace pv = epics::pvData;
+
+class ConvertTDCTest : public ::testing::Test {
+public:
+  virtual void SetUp() { TestConverter = TdcTime::Converter(); };
+  TdcTime::Converter TestConverter;
+};
+
+template <typename scalarArrayType> pv::ScalarType GetElementType() {
+  if (std::is_same<scalarArrayType, pv::PVByteArray>::value) {
+    return pv::ScalarType::pvByte;
+  } else if (std::is_same<scalarArrayType, pv::PVUByteArray>::value) {
+    return pv::ScalarType::pvUByte;
+  } else if (std::is_same<scalarArrayType, pv::PVShortArray>::value) {
+    return pv::ScalarType::pvShort;
+  } else if (std::is_same<scalarArrayType, pv::PVUShortArray>::value) {
+    return pv::ScalarType::pvUShort;
+  } else if (std::is_same<scalarArrayType, pv::PVIntArray>::value) {
+    return pv::ScalarType::pvInt;
+  } else if (std::is_same<scalarArrayType, pv::PVUIntArray>::value) {
+    return pv::ScalarType::pvUInt;
+  } else if (std::is_same<scalarArrayType, pv::PVLongArray>::value) {
+    return pv::ScalarType::pvLong;
+  } else if (std::is_same<scalarArrayType, pv::PVULongArray>::value) {
+    return pv::ScalarType::pvULong;
+  } else if (std::is_same<scalarArrayType, pv::PVFloatArray>::value) {
+    return pv::ScalarType::pvFloat;
+  } else if (std::is_same<scalarArrayType, pv::PVDoubleArray>::value) {
+    return pv::ScalarType::pvDouble;
+  } else {
+    assert(false);
+  }
+  return pv::ScalarType::pvInt;
+}
+
+template <typename scalarArrType>
+pv::PVStructure::shared_pointer CreateTestNTScalarArray(size_t elements) {
+  pvNT::NTScalarArrayBuilderPtr builder = pvNT::NTScalarArray::createBuilder();
+  pvNT::NTScalarArrayPtr ntScalarArray =
+      builder->value(GetElementType<scalarArrType>())->addTimeStamp()->create();
+
+  typename scalarArrType::svector someValues;
+  for (size_t i = 0; i < elements; i++) {
+    someValues.push_back(i * 2);
+  }
+
+  auto pvValueField = ntScalarArray->getValue<scalarArrType>();
+  pvValueField->replace(freeze(someValues));
+
+  pv::TimeStamp timeStamp;
+  timeStamp.fromTime_t(time(nullptr));
+  timeStamp += 0.123456789;
+  pv::PVTimeStamp pvTimeStamp;
+  ntScalarArray->attachTimeStamp(pvTimeStamp);
+  pvTimeStamp.set(timeStamp);
+
+  return ntScalarArray->getPVStructure();
+}
+
+pv::PVStructure::shared_pointer CreateTestScalarStruct() {
+  auto Builder = pvNT::NTScalar::createBuilder();
+  auto PVStruct = Builder->value(pv::pvDouble)->addTimeStamp()->create();
+  auto ValueField = PVStruct->getValue<pv::PVDouble>();
+  ValueField->put(3.14);
+  pv::TimeStamp TS;
+  TS.getCurrent();
+  pv::PVTimeStamp pvTS;
+  PVStruct->attachTimeStamp(pvTS);
+  return PVStruct->getPVStructure();
+}
+
+TEST_F(ConvertTDCTest, TwoElementSuccess) {
+  auto TestData = CreateTestNTScalarArray<pv::PVIntArray>(60);
+  FlatBufs::EpicsPVUpdate Update;
+  Update.epics_pvstr = TestData;
+  Update.channel = "somestring";
+  auto Result = TestConverter.create(Update);
+  EXPECT_NE(Result, nullptr);
+}
+
+TEST_F(ConvertTDCTest, OneElementFailure) {
```

Comment:
I think it would be helpful to future readers of this test to explain why it should fail when given a single element, it is only obvious to me because I read your PR description. The test name or a comment could be used.

It raises the question of what the behaviour is for 3 elements, does it throw away the last value and return a result based on the first pair, or fail in some way? Would be good to document it with another test.

Reply:
Added test and added documentation.